### ISSUE TITLE
False positive E1124 redundant-keyword-arg in complex subclassing scenario

### ISSUE THREAD

**Issue by cdeil** [PyCQA/pylint#2441]
This issue was first mentioned in https://github.com/PyCQA/pylint/issues/2335#issuecomment-415055075 and is similar.

I get a similar error when subclassing this class:

https://github.com/astropy/astropy/blob/ba8f644694d77865f7ba1376e7f0447fec5929b6/astropy/units/quantity.py#L196

With this example:
```
from astropy.units import Quantity


class Energy(Quantity):
    def __new__(cls, energy, unit=None, dtype=None, copy=True):
        self = super(Energy, cls).__new__(cls, energy, unit, dtype=dtype, copy=copy)
        return self
```
I get this error:
```
$ pylint -E test.py
************* Module test
test.py:6:15: E1124: Argument 'dtype' passed by position and keyword in classmethod call (redundant-keyword-arg)
```

The error makes no sense to me, on that line I pass `dtype=dtype` and it's not passed by position.

This is with astroid and pylint master from today.
```
$ pylint --version
pylint 2.2.0
astroid 2.0.4
Python 3.6.0 | packaged by conda-forge | (default, Feb 10 2017, 07:08:35) 
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)]
```

**Comment by clavedeluna** [comment]
Revisiting old issues. Trying to reproduce this with 
```
from astropy.units import Quantity


class Energy(Quantity):
    def __new__(cls, energy, unit=None, dtype=None, copy=True):
        self = super(Energy, cls).__new__(cls, energy, unit, dtype=dtype, copy=copy)
        return self
```

using pylint 2.16.0-dev and astropy-5.1.1, with no pylint config, we now get
```
test.py:4:0: W0223: Method 'choose' is abstract in class 'Quantity' but is not overridden in child class 'Energy' (abstract-method)
test.py:4:0: W0223: Method 'dump' is abstract in class 'Quantity' but is not overridden in child class 'Energy' (abstract-method)
test.py:4:0: W0223: Method 'dumps' is abstract in class 'Quantity' but is not overridden in child class 'Energy' (abstract-method)
test.py:4:0: W0223: Method 'tobytes' is abstract in class 'Quantity' but is not overridden in child class 'Energy' (abstract-method)
test.py:4:0: W0223: Method 'tofile' is abstract in class 'Quantity' but is not overridden in child class 'Energy' (abstract-method)
test.py:4:0: W0223: Method 'tolist' is abstract in class 'Quantity' but is not overridden in child class 'Energy' (abstract-method)
test.py:4:0: W0223: Method 'tostring' is abstract in class 'Quantity' but is not overridden in child class 'Energy' (abstract-method)
test.py:5:4: W0222: Signature differs from overridden '__new__' method (signature-differs)
```

which look pretty reasonable.

And to try to minimize the example, I linted
```
class Quantity:
    def __init__(self, energy, unit=None, dtype=None, copy=True):
        pass

class Energy(Quantity):
    def __init__(self, energy, unit=None, dtype=None, copy=True):
        super().__init__(energy, unit, dtype=dtype, copy=copy)
```

and no pylint messages. Though maybe this simplifies too much are removes the potential issue.

The original issue as stated isn't reproduced, so maybe this is worth closing?

**Comment by Pierre-Sassoulas** [comment]
Thank you for investigating and helping triage issue @clavedeluna !
